tests/unit/test_varlen_nan_detection.py

```
@@ -0,0 +1,173 @@
+    patterns["inf"] = g
+
+    return patterns
+
+
+def test_nan_gradient_detection():
+    """
+    Core test: NeuralDBG must detect NaN gradients regardless of source.
+    """
+    from neuraldbg import NeuralDbg
+
+    grad_norm, correct_grad = get_ground_truth()
```

Ruff (0.15.15)

[warning] Unpacked variable `grad_norm` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)